libweb3jsonrpc/Test.cpp

```diff
@@ -33,6 +34,60 @@ using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+Json::Value fillJsonWithState(eth::State const& _state)
+{
+    Json::Value oState;
+    for (auto const& a : _state.addresses())
+    {
+        Json::Value o;
+        o["balance"] = toCompactHexPrefixed(_state.balance(a.first), 1);
+        o["nonce"] = toCompactHexPrefixed(_state.getNonce(a.first), 1);
+
+        Json::Value store;
+        for (auto const& s : _state.storage(a.first))
+            store[toCompactHexPrefixed(s.second.first, 1)] =
+                toCompactHexPrefixed(s.second.second, 1);
+        o["storage"] = store;
+        o["code"] = toHexPrefixed(_state.code(a.first));
+        oState[toHexPrefixed(a.first)] = o;
+    }
+    return oState;
+}
+
+string exportLog(eth::LogEntries const& _logs)
+{
+    RLPStream s;
+    s.appendList(_logs.size());
+    for (eth::LogEntry const& l : _logs)
+        l.streamRLP(s);
+    return toHexPrefixed(sha3(s.out()));
+}
+
+const string c_postStateJustHashVersion = "justhash";
+const string c_postStateFullStateVersion = "fullstate";
+const string c_postStateLogHashVersion = "justloghash";
+string Test::test_getPostState(Json::Value const& param1)
+{
+    Json::Value out;
+    Json::FastWriter fastWriter;
+    if (param1["version"].asString() == c_postStateJustHashVersion)
+        return toJS(m_eth.postState().state().rootHash());
+    else if (param1["version"].asString() == c_postStateFullStateVersion)
+    {
+        out = fillJsonWithState(m_eth.postState().state());
+        return fastWriter.write(out);
+    }
+    else if (param1["version"].asString() == c_postStateLogHashVersion)
+    {
+        if (m_eth.blockChain().receipts().receipts.size() != 0)
+            return exportLog(m_eth.blockChain().receipts().receipts[0].log());
```

Comment:
This gets the receipt of only the first transaction of the block, is this your intention?

Reply:
Couldn't you just use receipts root which is in the block header?